### ISSUE TITLE
difference example

### ISSUE THREAD

**Issue by godaygo** [kriadmin/30-seconds-of-python-code#19]
The provided example is unnecessary complicated! `set`s in Python by themselves  define set operations like difference and many others...So it is better to replace this example:
```python
def difference(a, b):
    b = set(b)
    return [item for item in a if item not in b]
```
with this:
```python
def difference(lst1, lst2):
    return list(set(lst1) - set(lst2))
```
Also none of these examples take into account if elements of `a` or `b` (`lst1` or `lst2`) are hashable. The same critique goes for difference_by example.

**Comment by huybery** [comment]
Yes, I have the same idea as you. 😸

**Comment by kriadmin** [comment]
If you want to make a PR please make it soon because I will busy for this month and will not be able to respond to PRs and Isssues.

**Comment by kriadmin** [comment]
Fixed by #23
